tests/models/ReadActivity.test.js

```diff
@@ -0,0 +1,167 @@
+        href: 'http://example.org/abc',
+        hreflang: 'en',
+        mediaType: 'text/html',
+        name: 'An example link'
+      },
+      {
+        '@context': 'https://www.w3.org/ns/activitystreams',
+        type: 'Link',
+        href: 'http://example.org/abc2',
+        hreflang: 'en',
+        mediaType: 'text/html',
+        name: 'An example link2'
+      }
+    ]
+  }
+
+  const publication = await Publication.createPublication(
+    createdReader,
+    simplePublication
+  )
+
+  const selectorJsonObject = {
+    selector: { property: 'value' },
+    json: { anotherProperty: 88 }
+  }
+
+  const selectorObject = {
+    selector: { property: 'someValue' }
+  }
+
+  let id
+
+  await tap.test('Create ReadActivity with selector and json', async () => {
+    let readActivity = await ReadActivity.createReadActivity(
+      urlToId(createdReader.id),
+      urlToId(publication.id),
+      selectorJsonObject
+    )
+
+    let lastReadActivity = await ReadActivity.query().findById(
+      urlToId(readActivity.id)
+    )
+
+    await tap.ok(readActivity)
+    await tap.equal(readActivity.id, lastReadActivity.id)
+    await tap.equal(readActivity.readerId, createdReader.id)
+    await tap.equal(readActivity.publicationId, publication.id)
+    await tap.equal(readActivity.selector.property, 'value')
+    await tap.equal(readActivity.json.anotherProperty, 88)
+  })
+
+  await tap.test('Create ReadActivity with selector only', async () => {
+    let readActivity = await ReadActivity.createReadActivity(
+      urlToId(createdReader.id),
+      urlToId(publication.id),
+      selectorObject
+    )
+
+    await tap.ok(readActivity)
+    await tap.equal(readActivity.readerId, createdReader.id)
+    await tap.equal(readActivity.publicationId, publication.id)
+    await tap.equal(readActivity.selector.property, 'someValue')
+    await tap.equal(readActivity.json, null)
+  })
+
+  await tap.test('Create ReadActivity with non-existent readerId', async () => {
+    let readActivity = await ReadActivity.createReadActivity(
+      urlToId(createdReader.id + 'RandomString'),
+      urlToId(publication.id),
+      selectorObject
+    )
+
+    await tap.ok(typeof readActivity, Error)
+    await tap.equal(readActivity.message, 'no reader')
+  })
+
+  await tap.test(
+    'Create ReadActivity with non-existent publicationId',
+    async () => {
+      let readActivity = await ReadActivity.createReadActivity(
+        urlToId(createdReader.id),
+        urlToId(publication.id + 'AnotherRandomString'),
+        selectorObject
+      )
+
+      await tap.ok(typeof readActivity, Error)
+      await tap.equal(readActivity.message, 'no publication')
+    }
+  )
+
+  await tap.test('Get latests ReadActivity of a publication', async () => {
+    const newReader = {
+      name: 'Latest Reader'
+    }
+    const random = crypto.randomBytes(13).toString('hex')
+    const latestReader = await Reader.createReader(
+      `auth0|foo${random}`,
+      newReader
+    )
```

Comment:
this is fine, but you could also have continued to use the user you created at the beginning of the file.